src/BaGetter.Core/Upstream/Clients/V2UpstreamClient.cs

```diff
@@ -45,7 +45,27 @@ public class V2UpstreamClient : IUpstreamClient, IDisposable
 
         _ngLogger = NullLogger.Instance;
         _cache = new SourceCacheContext();
-        _repository = Repository.Factory.GetCoreV2(new PackageSource(options.Value.PackageSource.AbsoluteUri));
+        var packageSource = new PackageSource(options.Value.PackageSource.AbsoluteUri);
+        if (options.Value.Authentication is { } auth)
+        {
+            switch (auth.Type)
+            {
+                case MirrorAuthenticationType.None:
+                    break;
+                case MirrorAuthenticationType.Basic:
+                    packageSource.Credentials = new PackageSourceCredential(
+                        packageSource.Source,
+                        auth.Username,
+                        auth.Password,
+                        true,
+                        null);
+                    break;
+                default:
+                    throw new NotSupportedException($"The authentication type {auth.Type} is not supported.");
```

Comment:
Add something like "...is not supported for V2 upstream sources." or "...by the V2 protocol" to make sure someone just reading the exception knows what's going on.

Reply:
I tested and it works for me.